gestalt/__init__.py

```diff
@@ -588,7 +585,7 @@ def add_vault_config_provider(
         if client_config['url'] == "":
             client_config['url'] = os.environ["VAULT_ADDR"]
         if client_config['token'] == "":
-            client_config['token'] == os.environ["VAULT_TOKEN"]
+            client_config['token'] == os.environ.get("VAULT_TOKEN")
```

Comment:
@tanishq-dubey 
For type safety should `os.environ.get("VAULT_TOKEN")` > `os.environ.get("VAULT_TOKEN","")`